src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -20,28 +20,31 @@ namespace Microsoft.OData.UriParser
     /// A representation of the path portion of an OData URI which is made up of <see cref="ODataPathSegment"/>s.
     /// </summary>
     [SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "ODataPathCollection just doesn't sound right")]
-    public class ODataPath : IEnumerable<ODataPathSegment>
+    public class ODataPath : List<ODataPathSegment>, IEnumerable<ODataPathSegment>

```

Comment:
@marabooy Just curious, by having `OData` path inherit from `List<ODataPathSegment>` here, might we end up inadvertently surfacing behaviour available to `List<ODataPathSegment>` that is not applicable/relevant to `ODataPath`?

Reply:
For this maybe i could use a backing list inside to avoid polluting the api this way. 👍